File: my_cpp_pkg/src/custom.cpp
#include "rclcpp/rclcpp.hpp"
#include <iostream>

class MyNode_ : public rclcpp::Node {
public:
  MyNode_() : Node("this_is_a_really_long_name_for_a_cpp_node_but_it_works") {
    RCLCPP_INFO(this->get_logger(), "Hi");
  }

private:
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<MyNode_>();
  rclcpp::spin(node);
  rclcpp::shutdown();

  return 0;
}

File: my_py_pkg/my_py_pkg/custom.py
#!/usr/bin/env python3


import rclpy
from rclpy.node import Node

class MyNode_(Node):

    def __init__(self):
        super().__init__("custom_py")
        self.get_logger().info("Hi")

def main(arg = None):
    rclpy.init(args = arg)
    node = MyNode_()
    rclpy.spin(node)
    rclpy.shutdown()
if __name__ == "__main__":
    main()
